src/ui/public/state_management/state_monitor.js

```diff
@@ -19,7 +19,7 @@ function stateMonitor(state, defaultState) {
   }
 
   function getStatus() {
-    const currentState = filterState(state.toJSON());
+    const currentState = filterState(cloneDeep(state.toJSON()));
```

Comment:
Could you please add a comment here which explains the intent? Basically a terser version of this PR's description: "Get a copy of the underlying state's objects instead of the reference returned by toJSON(), because we need to mutate this object, e.g. in `filterState`."